h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoostUtils.java

```diff
@@ -163,17 +164,11 @@ private static int sumChunksLength(int[] chunkIds, Vec vec, Vec weightsVector) {
         int totalChunkLength = 0;
 
         for (int chunk : chunkIds) {
-            totalChunkLength += vec.chunkLen(chunk);
-
-            if(weightsVector == null) continue;
-            Chunk weightVecChunk = weightsVector.chunkForChunkIdx(chunk);
-            if (weightVecChunk.atd(0) == 0) totalChunkLength--;
-            int nzIndex = 0;
-            do {
-                nzIndex = weightVecChunk.nextNZ(nzIndex, true);
-                if (nzIndex < 0 || nzIndex >= weightVecChunk._len) break;
-                if (weightVecChunk.atd(nzIndex) == 0) totalChunkLength--;
-            } while (true);
+            if (weightsVector == null) {
+                totalChunkLength += vec.chunkLen(chunk);
+            } else {
+                totalChunkLength += countNonZerosInChunk(weightsVector.chunkForChunkIdx(chunk));
```

Comment:
This part is not directly related to the issue. However, counting zeros in a chunk emerged on yet another place, so the code has been extracted and only called from this method.

If there are no weights, let's just add the size of the whole chunk to the overall length. If there are weights present, rows with weights of zero should not be counted.